azure/template.json

```diff
@@ -0,0 +1,163 @@
+        "appServiceName": "[concat(variables('resourceNamePrefix'), '-as')]"
+    },
+    "resources": [
+        {
+            "apiVersion": "2017-05-10",
+            "name": "app-insights",
+            "type": "Microsoft.Resources/deployments",
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "[concat(variables('deploymentUrlBase'),'application-insights.json')]",
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
+                    "appInsightsName": {
+                        "value": "[variables('appServiceName')]"
+                    },
+                    "attachedService": {
+                        "value": "[variables('appServiceName')]"
+                    }
+                }
+            },
+            "dependsOn": [
+            ]
+        },
+        {
+            "apiVersion": "2017-05-10",
+            "name": "app-service-certificate",
+            "resourceGroup": "[parameters('sharedAppServicePlanResourceGroup')]",
+            "type": "Microsoft.Resources/deployments",
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "[concat(variables('deploymentUrlBase'),'app-service-certificate.json')]",
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
+                    "keyVaultCertificateName": {
+                        "value": "[parameters('certificateName')]"
+                    },
+                    "keyVaultName": {
+                        "value": "[parameters('sharedKeyVaultName')]"
+                    },
+                    "keyVaultResourceGroup": {
+                        "value": "[parameters('sharedKeyVaultResourceGroupName')]"
+                    }
+                }
+            }
+        },
+        {
+            "apiVersion": "2017-05-10",
+            "name": "app-service",
+            "type": "Microsoft.Resources/deployments",
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "[concat(variables('deploymentUrlBase'),'app-service.json')]",
+                    "contentVersion": "1.0.0.0"
+                },
+                "parameters": {
+                    "appServiceName": {
+                        "value": "[variables('appServiceName')]"
+                    },
+                    "appServicePlanName": {
+                        "value": "[parameters('sharedFrontendAppServicePlanName')]"
+                    },
+                    "appServicePlanResourceGroup": {
+                        "value": "[parameters('sharedAppServicePlanResourceGroup')]"
+                    },
+                    "appServiceAppSettings": {
```

Comment:
Could the extra application settings used in https://github.com/SkillsFundingAgency/das-reservations-api/blob/master/azure/template.json#L166-L197 be used, worth discussion whether the Redis connection string should also be within the application settings

Reply:
Doing a search across the project it looks like:
- EnvironmentName is used, not Environment.
- ConfigurationStorageConnectionString is already in there
- ConfigNames is not used
- Version is not used
- APPINSIGHTS_INSTRUMENTATIONKEY is in already
- ASPNETCORE_ENVIRONMENT is in already
- LoggingRedisConnectionString is in the ConnectionStrings section as Redis

Are ConfigNames and Version something that is widely used across DAS that should be in the project?

Reply:
All sounds good,

Having parameters for ConfigNames and Version to select the rowkey in the config table is preferable to hardcoding it like in https://github.com/SkillsFundingAgency/das-assessor-service/blob/master/src/SFA.DAS.AssessorService.Web/Startup.cs#L35-L36

Reply:
I can see that, but wouldn't it introduce the possibility of drift between what config version and name the code has set and expects v what is set during deployment?

Reply:
Those are added in.

Reply:
These have been added in.